src/cpp/core/system/Win32System.cpp

```diff
@@ -335,14 +336,29 @@ FilePath userSettingsPath(const FilePath& userHomeDirectory,
          appNameWide.c_str(),
          path);
 
-   if (hr != S_OK)
+   if (hr == S_OK)
+      return FilePath(std::wstring(path));
+
+   std::string localAppData = core::system::getenv("LOCALAPPDATA");
```

Comment:
Should we use an API for this versus relying on the env-var?

`SHGetKnownFolderPath(FOLDERID_LocalAppData, ...)`

Or better perhaps to use indirectly via `core::system::xdg::userDataDir()`?

Reply:
I don't think we want `xdg::userDataDir()` in this scenario, just because it would then also try to read some of the other XDG environment variables when resolving the path (and so would be a change in behavior). Using `SHGetKnownFolderPath` seems like a good idea, though.

Reply:
Reading through the lines a bit, the CSIDL enums are evidently deprecated:

https://learn.microsoft.com/en-us/windows/win32/shell/csidl

and APIs like https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetfolderpatha are considered deprecated, with preference to use the variants accepting the FOLDERID enums.

So I wonder if the best solution here is to eschew the use of `SHGetFolderPathAndSubDirW`, and instead just use `SHGetKnownFolderPath`, and complete from that path ourselves?